internal/tlsstate/controlmsg.go

```diff
@@ -0,0 +1,136 @@
+package tlsstate
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+
+	"github.com/ooni/minivpn/internal/bytesx"
+	"github.com/ooni/minivpn/internal/model"
+	"github.com/ooni/minivpn/internal/session"
+)
+
+// encodeClientControlMessage returns a byte array with the payload for a control channel packet.
```

Comment:
it seems conceptually a bit odd that we've got functions to deal with serialization and deserialization of control channel messages here at this layer.

however, I understand why we did this: at the moment, we're only using the TLS session for negotiating the keys (and then we destroy it). additionally, the tls state service is the only consumer of the control channel service, so it is natural to have these functions close to where we use them; that's it, in the doTLSAuth() algorithm.

I suspect we'll need to revisit this at some point (for one, I think we'll need to keep the tls session for longer, to be able to read possible incoming payloads being pushed by the server over the control channel; this is something we're neglecting for the time being); but for now this gets the job done and it fits our purposes of getting a usable data channel with minimal complexity.

In case we prefer to keep the control message serialization routines within the boundaries of the control channel package tho, we'll need to change the interfaces. I'm also not convinced that the name of the channels (and of this package in general) is very good tho - I renamed them to TLSRecordUp and TLSRecordDown but that's somehow of a stretch: yes, there are tls records *in the payloads*, but in reality it's control packets we're moving around.

Reply:
Would it be cleaner if the TLS channel was sending up and down raw records and the framing was instead implemented inside the control channel?

Reply:
I've looked at this a bit more, and I think I keep being biten by the terminology. A control *message* deals with payloads **over** the control channel; i.e., over the TLS conn that we get after the TLS handshake. We had implemented it correctly indeed: control channel passes us only the payload of CONTROL_V1 packets; and it also creates CONTROL_V1 **packets** moving down. 

At this layer what we're concerned about is control **messages**, and so the methods in `controlmsg` are probably fine here (since this is the only place at which we write over the control channel). The point about persisting the TLS session a bit more is a different one (i.e., who's the user of the TLS session), but I say we deal with that when we need.

On a minor note, perhaps renaming this package to `tlsession` is a good choice, since we do not only deal with state in here.

Reply:
Okay, then, let's rename the package!

Reply:
renamed!